### PR TITLE
Make `--dump-build-time option to print info in the console and hide the option to the end user

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/33582
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/33599

## Approach
```
$ bal build --dump-build-time                                                                                  
Compiling source
	pramodya/woo:0.1.0

Generating executable
	target/bin/woo.jar

Dumping build time information
	target/build-time.json
	timestamp : 1635931601841
	offline : false
	compile : false
	projectLoadDuration : 618
	packageResolutionDuration : 211
	packageCompilationDuration : 200
	codeGenDuration : 208
	emitArtifactDuration : 444
	testingExecutionDuration : 0
	totalDuration : 1712
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/DumpBuildTimeTask.java [hunk 1]
```diff
@@ -79,6 +81,19 @@ private Path getBuildTimeFilePath(Project project) {
         if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
             return project.sourceRoot().resolve("target").resolve(BUILD_TIME_JSON).toAbsolutePath();
         }
-        return currentDir.resolve("build-time.json").toAbsolutePath();
+        return currentDir.resolve(BUILD_TIME_JSON).toAbsolutePath();
+    }
+
+    private void printBuildTime(BuildTime buildTime) {
+        this.out.println("\ttimestamp : " + buildTime.timestamp);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/DumpBuildTimeTask.java:88]: Shall we avoid hardcoding the properties? You can use `gson.fromJson` to convert it to a `JsonObject`  and then iterate the map of entries

Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/DumpBuildTimeTask.java:88]: E.g. `JsonObject obj = new Gson().fromJson(BuildTime.getInstance(), JsonObject.class)`

Comment (pramodya1994) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/DumpBuildTimeTask.java]: Added

**Review by azinneera — APPROVED** [review]
LGTM

**Pull request merged**